src/workerd/api/crypto/scrypt.c++

```diff
@@ -18,20 +24,17 @@ kj::Maybe<jsg::BufferSource> scrypt(jsg::Lock& js,
     uint32_t maxmem,
     kj::ArrayPtr<const kj::byte> pass,
     kj::ArrayPtr<const kj::byte> salt) {
-  ClearErrorOnReturn clearErrorOnReturn;
-  auto buf = jsg::BackingStore::alloc<v8::ArrayBuffer>(js, length);
-  if (!EVP_PBE_scrypt(pass.asChars().begin(), pass.size(), salt.begin(), salt.size(), N, r, p,
-          maxmem, buf.asArrayPtr().begin(), length)) {
-    // This does not currently handle the errors in exactly the same way as
-    // the Node.js implementation but that's probably ok? We can update the
-    // error thrown to match Node.js more closely later if necessary. There
-    // are lots of places in the API currently where the errors do not match.
-    if (clearErrorOnReturn.peekError()) {
-      throwOpensslError(__FILE__, __LINE__, "crypto::scrypt");
-    }
-    return kj::none;
+  ncrypto::ClearErrorOnReturn clearErrorOnReturn;
+  auto backing = jsg::BackingStore::alloc<v8::ArrayBuffer>(js, length);
+  ncrypto::Buffer<kj::byte> buf(backing.asArrayPtr().begin(), backing.size());
+  ncrypto::Buffer<const char> passbuf{
+    .data = reinterpret_cast<const char*>(pass.begin()),
```

Comment:
is there a method that just returns const char*? I remember something like asChars().begin()?